### PR TITLE
TSV-537: Whitelist absolute and protocol-relative URLs

### PR DESCRIPTION
**Pull request by cellis-ee**

### CODE REVIEW

File: app/config/AppConfig.scala [hunk 1]
```diff
@@ -22,11 +22,16 @@ import uk.gov.hmrc.play.config.ServicesConfig
 trait AppConfig {
   def platformFrontendHost: String
   def emailServicePath: String
+  def whitelistedDomains: Set[String]
 }
 
 object AppConfig extends AppConfig with ServicesConfig {
   override val platformFrontendHost = getConfigValueFor("platform.frontend.host")
```
Comment (anshulbajpai) [inline app/config/AppConfig.scala]: Move the behaviour from this object to trait. This will be inline with  our other microservices

Comment (jachro) [reply app/config/AppConfig.scala]: 👍

File: app/uk/gov/hmrc/emailverification/controllers/ForwardUrl.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    override def reads(json: JsValue): JsResult[ForwardUrl] = {
+      val url = json.as[String]
+      validate(url) match {
+        case Left(message) => JsError(error = message)
+        case Right(forwardUrl) => JsSuccess(forwardUrl)
+      }
+    }
+  }
+
+  def validate(potentialUrl: String): Either[String, ForwardUrl] = {
```
Comment (jachro) [inline app/uk/gov/hmrc/emailverification/controllers/ForwardUrl.scala]: private maybe?

**Review by jachro — CHANGES_REQUESTED** [review]

File: test/uk/gov/hmrc/emailverification/services/VerificationLinkServiceSpec.scala [hunk 3]
```diff
@@ -33,7 +33,7 @@ class VerificationLinkServiceSpec extends UnitSpec with MockitoSugarRush {
       val continueUrl = "http://continue-url.com"
       val expectedExpirationTimeStamp = ""
 
-      val verificationRequest = EmailVerificationRequest(emailToVerify, templateId, templateParams, Period.parse("P1D"), continueUrl)
+      val verificationRequest = EmailVerificationRequest(emailToVerify, templateId, templateParams, Period.parse("P1D"), ForwardUrl(continueUrl))
```
Comment (jachro) [inline test/uk/gov/hmrc/emailverification/services/VerificationLinkServiceSpec.scala]: Not used

File: test/uk/gov/hmrc/emailverification/services/VerificationLinkServiceSpec.scala [hunk 4]
```diff
@@ -33,7 +33,7 @@ class VerificationLinkServiceSpec extends UnitSpec with MockitoSugarRush {
       val continueUrl = "http://continue-url.com"
```
Comment (jachro) [inline test/uk/gov/hmrc/emailverification/services/VerificationLinkServiceSpec.scala]: 'continueUrl' should be a ForwardUrl.

**Review by jachro — APPROVED** [review]

**Comment by cellis-ee** [comment]
:+1: from @jachro

**Comment by cellis-ee** [comment]
:+1: from @dclifton-hmrc

**Pull request merged**